Implement Jira connector with test cases (#5)
ballerina-platform/module-ballerinax-jira: README.md

The `ballerinax/jira` package offers APIs to interact with the [Jira REST API](https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro), providing access to various endpoints for managing issues, filters, projects, and other resources.

## Setup guide

To use the Jira connector, you must have a Jira Cloud account with API access. If you do not have one, you can sign up for a Jira account on [Atlassian’s website](https://developer.atlassian.com/cloud/jira/platform/).

### Step 1: Create an API Token
1. Login to [Atlassian Account](https://developer.atlassian.com/cloud/jira/platform/)

2. Click on the profile icon to get into developer console

![{D8EEDDDF-5D02-4907-A10A-911BB513050F}](https://github.com/user-attachments/assets/3ec0f5be-6c5f-4038-a37a-ec8a2f6e2ccf)

3. In the new window, click on the profile again and go to 'Manage Account'

4. Click and go to security tab and scroll down to the bottom to generate API token by clicking 'Create and Manage API token'

![{8E9B6CE3-B218-46F4-8481-5211FE443418}](https://github.com/user-attachments/assets/081db352-e5d1-4657-8f85-49f791881975)

5. Give a name to the API token and copy and save it in a secret place

> ![{2385E238-124F-4D5C-9446-836EF0EF56A2}](https://github.com/user-attachments/assets/d76705c0-2bf9-4f04-9d9a-1ec963d950e9)

Reviewer: Could you please add these images in the folder path `docs/setup/resources` and refer them here?
You can refer how it is done in Twitter connector -> https://github.com/ballerina-platform/module-ballerinax-twitter/tree/main/docs/setup/resources

Author: Did this change

## Overview

[Jira](https://www.atlassian.com/software/jira)  is a widely-used issue and project tracking software by Atlassian. It provides a REST API that allows applications to connect with Jira to create, update, and delete various resources, enabling efficient project and task management.

The `ballerinax/jira` package offers APIs to interact with the [Jira REST API](https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro), providing access to various endpoints for managing issues, filters, projects, and other resources.

## Setup guide

To use the Jira connector, you must have a Jira Cloud account with API access. If you do not have one, you can sign up for a Jira account on [Atlassian’s website](https://developer.atlassian.com/cloud/jira/platform/).

### Step 1: Create an API Token

> 1. Login to [Atlassian Account](https://developer.atlassian.com/cloud/jira/platform/)

Reviewer: ```suggestion
1. Log in to [Atlassian Account](https://developer.atlassian.com/cloud/jira/platform/).
```

> 2. Click on the profile icon to get into developer console

Reviewer: ```suggestion
2. Click on the profile icon to get into the developer console.
```

>  <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/Atlassian-developer-console#1.png alt="Atlassian developer console" style="width: 70%;">

Reviewer: Usually we start the image names in a simple letter. ex: `atlassian-developer-console`
This is a nice to have one. :)

Author: @Bhashinee did the change

>  <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/atlassian-developer-console#1.png alt="Atlassian developer console" style="width: 70%;">

Reviewer: You have to rename the image in the resources folder to this.

Author: It is already renamed

> 3. In the new window, click on the profile again and go to 'Manage Account'

Reviewer: ```suggestion
3. In the new window, click on the profile again and go to `Manage Account`
```

> 4. Click and go to security tab and scroll down to the bottom to generate API token by clicking 'Create and Manage API token'

Reviewer: ```suggestion
4. Click and go to the security tab and scroll down to the bottom to generate an API token by clicking `Create and Manage API token`
```

>  <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/create-API-Token#2.png alt="Create API Token" style="width: 70%;">

Reviewer: Rename the image in the resources folder.

Author: I have renamed all the images

> 5. Give a name to the API token and copy and save it in a secret place

Reviewer: ```suggestion
5. Give a name to the API token and copy and save it in a secret place.
```

> <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/generate-API-Token#3.png alt="Generate API Token" style="width: 70%;">

Reviewer: Rename the image in the resource folder.

Author: I dont see a discrepancy with the image names in the resource folder as I have already changed the first letter of the images as lowercase

Reviewer: There is no image named `generate-API-Token#3.png` in the resources folder. The name that is there is `Generate-API-Token#3.png`. 
<img width="289" alt="Screenshot 2024-11-19 at 21 00 15" src="https://github.com/user-attachments/assets/324ca304-a2ee-4b49-9e73-8d4737924f95">

Reviewer: Check the other images as well

Author: Issue with the image name is it is not getting updated even though I have pushed the changed name. I just removed all image and added them and committed. Let's see

### Step 2: Integrate JIRA for your account

> 1. Click on the grid shaped icon in the right corner (Before Atlassian Account logo). Click on explore products

Reviewer: ```suggestion
1. Click on the grid shaped icon in the right corner (Before Atlassian Account logo). Click on `Explore products`.
```

> <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/try-jira-cloud#5.png alt="Try jira" style="width: 70%;">

Reviewer: Rename the image in the resource folder.

Reviewer: This image does not have the `Explore products` icon. It is there in the `try-jira-cloud` image. Could you please recheck if you have linked the correct images?

> 2. Click on the try cloud button under Jira

Reviewer: ```suggestion
2. Click on the try cloud button under Jira.
```

<img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/integrate-Jira#4.png alt="Integrate Jira" style="width: 70%;">

> 3.Click try jira for free

Reviewer: ```suggestion
3. Click `Get Jira for free`.
```

> <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/get-Jira-Free#6.png alt="Get jira for free" style="width: 70%;">

Reviewer: Rename the images in the resource folder.

4.Enter your email that you used previously to log into atlassian account

> <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/email-for-jira#7.png alt="Enter email for jira account" style="width: 70%;">

Reviewer: Rename the images in the resource folder.

> 5.Make sure to copy the Domain URL provided for you (Your site url)

Reviewer: ```suggestion
5. Make sure to copy the Domain URL provided for you (Your site URL)
```

> <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-jira/main/docs/setup/resources/domain-URL-Jira#8.png alt="Create API Token" style="width: 70%;">

Reviewer: Image names are different in the resource folder. Please check all the images.

Author: will do

> You have successfully integrated jira with your atlassian account. Now you can access the services through the API

Reviewer: ```suggestion
You have successfully integrated Jira with your Atlassian account. Now you can access the services through the API.
```

## Overview

> [Jira](https://www.atlassian.com/software/jira)  is a widely-used issue and project tracking software by Atlassian. It provides a REST API that allows applications to connect with Jira to create, update, and delete various resources, enabling efficient project and task management.

Reviewer: ```suggestion
[Jira](https://www.atlassian.com/software/jira) is a widely-used issue and project tracking software by Atlassian. It provides a REST API that allows applications to connect with Jira to create, update, and delete various resources, enabling efficient project and task management.
```
